### ISSUE TITLE
prefer-const reporting in try catch

### ISSUE THREAD

**Issue by edi9999** [babel/babel-eslint#495]
This issue comes from https://github.com/eslint/eslint/issues/8723

**Tell us about your environment**

* **ESLint Version:** 4.0.0
* **Node Version:** 8.1.0
* **npm Version:** 5.0.3

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**Please show your full configuration:**

```
{
  "parser": "babel-eslint",
  "plugins": [
    "dependencies"
  ],
  "rules": {
    "prefer-const": 2
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
let result;
try {
	result = window.bar();
}
catch (error) {
	console.log(error);
}
console.log(result);
```

**What did you expect to happen?**

No error to be thrown. It doesn't throw an error with 3.19.0

**What actually happened? Please include the actual, raw output from ESLint.**

```
/home/edgar/projects/dx/xt/es6/lexer.js
  3:2  error  'result' is never reassigned. Use 'const' instead  prefer-const
```

**Comment by hzoo** [comment]
And this doesn't fail with eslint without babel-eslint?

**Comment by edi9999** [comment]
Yes it doesnt fail with the default parser

**Comment by existentialism** [comment]
Have you tried:

```json
{
  "parserOptions" : {
    "ecmaVersion": 7
  },
  "parser": "babel-eslint",
  "rules": {
    "prefer-const": 2
  }
}
```

(Matches your default parser config you used in your original eslint issue)

**Comment by edi9999** [comment]
Indeed it does work with that config. 

Thanks
